common/persistence/health_signal_aggregator.go

```diff
@@ -103,14 +104,13 @@ func (s *HealthSignalAggregatorImpl) Stop() {
 }
 
 func (s *HealthSignalAggregatorImpl) Record(callerSegment int32, latency time.Duration, err error) {
-	// TODO: uncomment when adding dynamic rate limiter
-	//s.latencyAverage.Record(latency.Milliseconds())
-	//
-	//if isUnhealthyError(err) {
-	//	s.errorRatio.Record(1)
-	//} else {
-	//	s.errorRatio.Record(0)
-	//}
+	s.latencyAverage.Record(latency.Milliseconds())
+
+	if isUnhealthyError(err) {
+		s.errorRatio.Record(1)
+	} else {
+		s.errorRatio.Record(0)
+	}
```

Comment:
I am still a little bit concerned about the performance of this especially for cloud. Is it possible to add a separate flag (that requires a restart) to control if latency and error should be tracked? 

I understand this aggregator can be disabled, but we will also lose the hot shard metric. So I'd like to separate them.

Reply:
What is the performance concern? That we cannot keep up with the throughput?

I will add the flag for safety, but I would like to understand the specific concern.

Reply:
yeah basically if the windowed average impl can handle the throughput since all persistence will essentially get serialized when going through the aggregator and the operation is more complicated than simply counting the request.

well I do think it should be fine. Just want to play safe and make sure we have a way to disable part of it. The flag can be true by default.